langchain/src/agents/openai/output_parser.ts

```diff
@@ -0,0 +1,83 @@
+import type { OpenAI as OpenAIClient } from "openai";
+import {
+  AgentAction,
+  AgentFinish,
+  BaseMessage,
+  ChatGeneration,
+  Generation,
+  isBaseMessage,
+} from "../../schema/index.js";
+import { AgentActionOutputParser } from "../types.js";
+import { OutputParserException } from "../../schema/output_parser.js";
+
+/**
+ * Type that represents an agent action with an optional message log.
+ */
+export type FunctionsAgentAction = AgentAction & {
+  messageLog?: BaseMessage[];
+};
+
+export class OpenAIFunctionsAgentOutputParser extends AgentActionOutputParser {
+  lc_namespace = ["langchain", "agents", "openai"];
```

Comment:
Oh, for all of these we will want to add `static lc_name` to handle minification:

https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/chat_models/anthropic.ts#L151